pkgs/development/web/bun/default.nix

```diff
@@ -0,0 +1,19 @@
+{ stdenvNoCC, callPackage, lib }:
+let
+  version = "0.1.1";
+  pname = "bun";
+  meta = with lib; {
+    homepage = "https://bun.sh";
+    changelog = "https://github.com/Jarred-Sumner/bun/releases/tag/bun-${version}";
+    description = "Incredibly fast JavaScript runtime, bundler, transpiler and package manager – all in one.";
+    longDescription = ''
+      All in one fast & easy-to-use tool. Instead of 1,000 node_modules for development, you only need bun.
+    '';
+    license = licenses.mit;
+    maintainers = with maintainers; [ DAlperin ];
+    platforms = [ "x86_64-linux" "x86_64-darwin" "aarch64-darwin" ];
+  };
+in
+if stdenvNoCC.isDarwin
+then callPackage ./darwin.nix { inherit pname version meta; }
+else callPackage ./linux.nix { inherit pname version meta; }
```

Comment:
What if we call this from windows or bsd? Consider asserting or at least throwing an error